modules/mini-games/tightspot.lua

```diff
@@ -0,0 +1,758 @@
+            local offer = {price = {{"coin", tightspot_prices[item]}}, offer = {type = "give-item", item = item}}
+            market.add_market_item(offer)
+        end
+    end
+end
+local function start(args)
+    variables["level"] = {}
+    variables["surface"] = {}
+    variables["walls"] = {}
+    local level_index = args[1]
+    variables.level = config[level_index]
+    variables.diffuclty = variables.level.money[args[2]]
+    variables.loan_price =  variables.level.loan_prices[args[2]]
+    variables["surface"] = game.surfaces[variables.level["surface"]]
+    if not save["tiles"][1] then
+        level_save()
+    end
+    for i, player in ipairs(game.connected_players) do
+        player_join_game(player,i-1)
+    end
+    local force = game.players[1].force
+    force.disable_all_prototypes()
+    local recipe_list = force.recipes
+    for index, item in pairs(variables.level.recipes) do
+        recipe_list[item].enabled = true
+    end
+
+    market_setup()
+    variables.tick = game.tick
+end
+
+
+local function reset_table(table)
+    for i, _ in pairs(table) do
+        table[i] = nil
+    end
+end
+
+local function getSuffix (n)
+    local lastTwo, lastOne = n % 100, n % 10
+    if lastTwo > 3 and lastTwo < 21 then return "th" end
+    if lastOne == 1 then return "st" end
+    if lastOne == 2 then return "nd" end
+    if lastOne == 3 then return "rd" end
+    return "th"
+end
+
+local function Nth (n) return n..getSuffix(n) end
+
+local function stop()
+    game.speed = 1
+    for i,player in ipairs(game.connected_players) do
+        player.set_controller {type = defines.controllers.god}
+        player.create_character()
+        --gui
+        local Main_gui = Gui.get_left_element(player, game_gui)
+        local table = Main_gui.container["slider"].table
+        table.visible = false
+        Gui.toggle_left_element(player,  Main_gui, false)
+    end
+    started[1] = false
+
+    local area = variables.level.area
+    area[1][1] = area[1][1]
+    area[2][1] = area[2][1]
+    local left_overs = variables["surface"].find_entities_filtered {area= area}
+    for i, ent in ipairs(left_overs) do
+        if ent.name ~= "market" and ent.name ~= "steel-chest" then
+            ent.destroy()
+        end
+    end
+    for i, entity in ipairs(save["entity"]) do
+        local name = entity[1]
+        local position = entity[2]
+        local force = entity[3]
+        local minable = entity[4]
+        position.x = position.x
+        entity[2].x = position.x
+        local ent = variables["surface"].create_entity{name = name , position = position , force = force }
+        ent.minable = minable
+    end
+    local players = {}
+    for name,_ in pairs(centers) do
+        players[#players+1] = name
+    end
+    local scores = {}
+    for _,name in ipairs(players) do
+        local player = game.players[name]
+        scores[#scores+1] = {Store.get(balances,player),name}
+    end
```

Comment:
Why have two loops when it can be done with a single loop?
Also you dont need to get the player object to use Store, when you register the store you are telling it to use the player's name so if you pass the player name then it works just as well as a player object.
```suggestion
    local scores = {}
    for name in pairs(centers) do
        scores[#scores+1] = {Store.get(balances,name), name}
    end
```